services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -188,63 +178,54 @@ const updateBlockIndex = async job => {
-		const forkedTransactionIDs = await getTransactionsByBlockIDs(blocks.map(b => b.id));
-		await transactionsDB.deleteByPrimaryKey(forkedTransactionIDs, trx);
-		// TODO: Invoke revertTransaction method
-
-		// Update producedBlocks & rewards
-		await BluebirdPromise.map(
-			generatorPkInfoArray,
-			async pkInfoArray => {
-				await accountsDB.decrement({
-					decrement: {
-						rewards: BigInt(pkInfoArray.reward),
-						producedBlocks: 1,
-					},
-					where: {
-						property: 'address',
-						value: getBase32AddressFromPublicKey(pkInfoArray.publicKey),
-					},
-				}, trx);
-			});
-		await blocksDB.deleteByPrimaryKey(blocks.map(b => b.height), trx);
-		await commitDbTransaction(trx);
-		logger.debug(`Committed MySQL transaction to delete block(s) with ID(s): ${blockIDs}`);
+		if (block.transactions.length) {
+			const transactionsDB = await getTransactionsIndex();
+			const { transactions, assets, ...blockHeader } = block;
+
+			await BluebirdPromise.map(
+				transactions,
+				async (tx) => {
+					// Invoke 'revertTransaction' to execute command specific reverting logic
+					await revertTransaction(blockHeader, tx, dbTrx);
+
+					const forkedTransactionIDs = await getTransactionsByBlockIDs([blockHeader.id]);
+					await transactionsDB.deleteByPrimaryKey(forkedTransactionIDs, dbTrx);
+				},
+				{ concurrency: block.transactions.length },
+			);
+		}
+
+		await blocksDB.deleteByPrimaryKey(block.blockHeader.id);
+		await commitDbTransaction(dbTrx);
+		logger.debug(`Committed MySQL transaction to delete block with ID: ${block.blockHeader.id}`);
 	} catch (error) {
-		logger.debug(`Rolled back MySQL transaction to delete block(s) with ID(s): ${blockIDs}`);
-		await rollbackDbTransaction(trx);
+		logger.debug(`Rolled back MySQL transaction to delete block with ID: ${block.blockHeader.id}`);
+		await rollbackDbTransaction(dbTrx);
 
 		if (error.message.includes('ER_LOCK_DEADLOCK')) {
-			const errMessage = `Deadlock encountered while deleting block(s) with ID(s): ${blockIDs}. Will retry later.`;
+			const errMessage = `Deadlock encountered while deleting block with ID: ${block.blockHeader.id}. Will retry later.`;
 			logger.warn(errMessage);
 			throw new Error(errMessage);
 		}
 
-		logger.warn(`Error occured while deleting block(s) with ID(s): ${blockIDs}. Will retry later.`);
+		logger.warn(`Error occured while deleting block with ID: ${block.blockHeader.id}. Will retry later.`);
 		throw error;
 	}
 };
 
 // Initialize queues
 const indexBlocksQueue = Queue(config.endpoints.cache, 'indexBlocksQueue', indexBlock, 30);
 const updateBlockIndexQueue = Queue(config.endpoints.cache, 'updateBlockIndexQueue', updateBlockIndex, 1);
-const deleteIndexedBlocksQueue = Queue(config.endpoints.cache, 'deleteIndexedBlocksQueue', deleteIndexedBlocks, 1);
+const deleteIndexedBlocksQueue = Queue(config.endpoints.cache, 'deleteIndexedBlocksQueue', deleteIndexedBlock, 1);
```

Comment:
i guess the logic was to delete block which you have fixed but the job names still refer plural? if it make sense update them as well to be singular

Reply:
or was it intended to achieve multiple block deletion?

Reply:
Good point. It is intended to support multiple block deletion. So, I updated the implementation now